### PR TITLE
Refactor project to TypeScript

### PR DESCRIPTION
**Pull request by Borewit**
- Migrate project to TypeScript
- Enable (TypeScript) code linter

Solves / superseeds issue: #7

### CODE REVIEW

File: src/cue.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  const track = cuesheet.getCurrentTrack();
+
+  if (!track) {
+    throw new Error('No track for adding flag: ' + params);
+  }
+
+  track.flags = params.slice(0);
+}
+
+export function parseIndex(params: string[], cueSheet: CueSheet): Index {
```
Comment (justlaputa) [inline src/cue.ts]: do we have to `export` these internal `parseXXX` functions? I think no place use them right?

File: src/cue.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    track.title = params[0];
+  }
+}
+
+function parseTrack(params, cuesheet: CueSheet) {
+  const _number = parseInt(params[0], 10);
+  cuesheet.newTrack(_number, params[1]);
+}
+
+export interface ITime {
```
Comment (justlaputa) [inline src/cue.ts]: I think we can move this interface into `cuesheet.ts`, and import from it.

File: src/cue.ts (hunk 1 above)
Comment (Borewit) [reply src/cue.ts]: Indeed, thx

File: src/cue.ts (hunk 2 above)
Comment (Borewit) [reply src/cue.ts]: I moved all interfaces to a dedicated file, that can be benefitial for importing the TypeScript defenitions in a browser module. Which become related to #3 (supporting stream), I may have look if I can split the parsing functions for Node and browser streams to offer browser compatibility.

Comment (justlaputa) [reply src/cue.ts]: great, thank you!

**Pull request merged**